Checkpoint-Profile-Lookup.md

```diff
@@ -59,7 +60,24 @@ for (var x = 0; x < contacts.length; x++){
 return "No such contact";
 ```
 
+#### Second solution
+(Declarative programming)
+
+```
+  var result = "No such contact";
+  contacts.map(function (contact) {
```

Comment:
When you are not modifying the contents of the calling array (`contacts`), you should use `forEach`.

I don't really like this solution because it forces you to loop through all of the contacts even if you find the contact.  Imagine that you had 10000 contacts and the answer was in the first one?  What a waste that would be.  This is one of the rare cases where functional programming doesn't make sense.

You could use `.some` instead:
```
contacts.some(funciton(contact) {
    if (contact.firstName == firstName) {
      if (contact[prop]){
        result = contact[prop];
      }
     else result = "No such property";
    return true;  // Exit the loop
    }
   return false; // Keep going
});
return result;
}